indico/modules/rb_new/client/js/modules/landing/UpcomingBookings.jsx

```diff
@@ -0,0 +1,105 @@
+/* This file is part of Indico.
+ * Copyright (C) 2002 - 2018 European Organization for Nuclear Research (CERN).
+ *
+ * Indico is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {bindActionCreators} from 'redux';
+import {Header, List, Popup, Label} from 'semantic-ui-react';
+import {connect} from 'react-redux';
+import {toMoment} from 'indico/utils/date';
+import SpriteImage from '../../components/SpriteImage';
+import {actions as bookingActions} from '../../common/bookings';
+
+import './UpcomingBookings.module.scss';
+
+
+/**
+ * A widget that shows the user a list of their upcoming bookings
+ */
+class UpcomingBookings extends React.Component {
+    static propTypes = {
+        /** The list of booking objects */
+        bookings: PropTypes.arrayOf(PropTypes.object).isRequired,
+        actions: PropTypes.shape({
```

Comment:
`.exact()` instead of `.shape()` so it shows a proptypes error if we ever add another action without adding it here